### ISSUE TITLE
Passing CMAKE_INSTALL_LIBDIR breaks GNUInstallDirs

### ISSUE THREAD

**Issue by jtojnar** [Haivision/srt#538]
On Nix, we are passing the following flags: `-DCMAKE_INSTALL_INCLUDEDIR=/nix/store/ij8r0s1nzhm9x2c3z3jcziq7r7bh2h0s-srt-1.3.1-dev/include -DCMAKE_INSTALL_LIBDIR=/nix/store/vs6d2fjkl4kb3jb7rwibsd76k9v2n4xy-srt-1.3.1/lib -DCMAKE_INSTALL_NAME_DIR=/nix/store/yvlj5bph7d8cyx9ha78vvhxngnvi3zcw-srt-1.3.1/lib -DCMAKE_POLICY_DEFAULT_CMP0025=NEW -DCMAKE_OSX_DEPLOYMENT_TARGET= -DCMAKE_OSX_SYSROOT= -DCMAKE_FIND_FRAMEWORK=last -DCMAKE_STRIP=/nix/store/3xwc1ip20b0p68sxqbjjll0va4pv5hbv-binutils-2.30/bin/strip -DCMAKE_RANLIB=/nix/store/3xwc1ip20b0p68sxqbjjll0va4pv5hbv-binutils-2.30/bin/ranlib -DCMAKE_AR=/nix/store/3xwc1ip20b0p68sxqbjjll0va4pv5hbv-binutils-2.30/bin/ar -DCMAKE_C_COMPILER=gcc -DCMAKE_CXX_COMPILER=g++ -DCMAKE_INSTALL_PREFIX=/nix/store/yvlj5bph7d8cyx9ha78vvhxngnvi3zcw-srt-1.3.1 `

Unfortunately, when [`GNUInstallDirs`](https://cmake.org/cmake/help/latest/module/GNUInstallDirs.html)’s `CMAKE_INSTALL_LIBDIR` is specified, `GNUInstallDirs` gets disabled, breaking the installation.

**Comment by ethouris** [comment]
Yes, overriding just `CMAKE_INSTALL_LIBDIR` causes that the values of all `CMAKE_INSTALL_*` variables are completely up to you and `GNUInstallDirs` are not in use. We can fix it by checking the specification of all variables, but the only difference here will be that you'll get error directly from cmake. Please follow the statement in the `CMakeLists.txt`:

```
# (...). Shall this be not desired, override values
# of CMAKE_INSTALL_BINDIR, CMAKE_INSTALL_LIBDIR and CMAKE_INSTALL_INCLUDEDIR.
```
If you want to let `GNUInstallDirs` work, you can simply specify only `-DCMAKE_INSTALL_PREFIX=/nix/store/ij8r0s1nzhm9x2c3z3jcziq7r7bh2h0s-srt-1.3.1-dev`.

**Comment by jtojnar** [comment]
Sorry for not replying sooner, somehow I did not get the message.

The issue is that the `libdir` and `includedir` use different prefix so `-DCMAKE_INSTALL_PREFIX=/nix/store/ij8r0s1nzhm9x2c3z3jcziq7r7bh2h0s-srt-1.3.1-dev` is not sufficient.

**Comment by ethouris** [comment]
What's the problem then with using separate variables for separate path types?

**Comment by jtojnar** [comment]
Sorry, somehow this has evaded my attention again.

I think the issue was that we wanted to set custom `libdir` and keep the default `bindir` and `includedir`.

I do not really see why `GNUInstallDirs` are disabled when `CMAKE_INSTALL_LIBDIR` is passed. It still allows overriding the values individually when needed.

**Comment by ethouris** [comment]
This could be fixed by requiring to specify the option through custom variables, likely such as `WITH_INSTALL_LIBDIR`. This would then still include `GNUInstallDirs`, but if `WITH_INSTALL_*` options were defined, they will override `CMAKE_INSTALL_*`.
